File: write-number-in-expanded-form/kata.js
function expandedForm(num) {
  num = String(num).split('');
  result = [];
  num.map((number, indeks) => {
    if (number != '0') {
      result.push(number * Math.pow(10, num.length - 1 - indeks));
    }
  });
  return result.join(' + ');
}

function expandedForm(num) {
  num = String(num).split('');
  const newNum = num.map((val, i) => val != 0 && String(val * Math.pow(10, num.length - 1 - i))).filter(Boolean);
  return newNum.join(' + ');
}

console.log(expandedForm(30704));

File: write-number-in-expanded-form/kata.py
from math import pow

def expanded_form(num):
  numStr = str(num)
  numArr = list(numStr)
  result = []
  for indeks, number in enumerate(numArr):
    if number != '0':
      result.append(str(int(number) * int(pow(10, len(numArr) - 1 - indeks))))
  return ' + '.join(result)

def expanded_form(num):
  num = str(num)
  return ' + '.join(x + '0' * (len(num) - i - 1) for i, x in enumerate(num) if x != '0')

print(expanded_form(20304))  # Outputs: '20000 + 300 + 4'